besspin/cyberPhys/cyberphyslib/cyberphyslib/demonstrator/can_out.py

```diff
@@ -39,20 +40,46 @@ def on_poll_poll(self, t):
         if self._location is not None:
             pos = self._location[0]
 
-            # Send position X, Y 
-            self._network.send(canspecs.CAN_ID_CAR_X, struct.pack("!f", pos[0]))
-            self._network.send(canspecs.CAN_ID_CAR_Y, struct.pack("!f", pos[1]))
-            self._network.send(canspecs.CAN_ID_CAR_Z, struct.pack("!f", pos[2]))
+            # Send position X, Y, Z
+            if self._last_location is not None:
+                last_pos = self._last_location[0]
+                send_x = abs(last_pos[0] - pos[0]) > config.MIN_POSITION_CHANGE
+                send_y = abs(last_pos[1] - pos[1]) > config.MIN_POSITION_CHANGE
+                send_z = abs(last_pos[2] - pos[2]) > config.MIN_POSITION_CHANGE
+            else:
+                send_x = True
+                send_y = True
+                send_z = True
```

Comment:
Not important, but a tuple unpack might simpler
```suggestion
            if self._last_location is not None:
                last_pos = self._last_location[0]
                send_x, send_y, send_z = [abs(lpi - pi) > config.MIN_POSITION_CHANGE for lpi, pi in zip(last_pos, pos)]
            else:
                send_x, send_y, send_z = (True, True, True)
```

Reply:
Seems reasonable enough to me.